sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/transaction/TransactionEncoderImpl.kt

```diff
@@ -133,48 +208,53 @@ internal class TransactionEncoderImpl(
-            // TODO [#1359]: Expose the proposal in a way that enables querying its fee.
-            // TODO [#1359]: https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/issues/1359
             val proposal =
                 backend.proposeTransfer(
-                    usk,
+                    usk.account,
                     toAddress,
                     amount.value,
                     memo
                 )
-            backend.createProposedTransaction(proposal, usk)
-        } catch (t: Throwable) {
-            Twig.debug(t) { "Caught exception while creating transaction." }
-            throw t
-        }.also { result ->
+            val transactionIds = backend.createProposedTransactions(proposal, usk)
+            assert(transactionIds.size == 1)
+            transactionIds[0]
+        }.onFailure {
+            Twig.error(it) { "Caught exception while creating transaction." }
+        }.onSuccess { result ->
             Twig.debug { "result of sendToAddress: $result" }
-        }
+        }.getOrThrow()
     }
 
     private suspend fun createShieldingSpend(
         usk: UnifiedSpendingKey,
         memo: ByteArray? = byteArrayOf()
     ): FirstClassByteArray {
-        @Suppress("TooGenericExceptionCaught")
-        return try {
+        return runCatching {
             saplingParamTool.ensureParams(saplingParamTool.properties.paramsDirectory)
             Twig.debug { "params exist! attempting to shield..." }
-            // TODO [#1359]: Expose the proposal in a way that enables querying its fee.
-            // TODO [#1359]: https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/issues/1359
-            val proposal = backend.proposeShielding(usk, memo)
-            backend.createProposedTransaction(proposal, usk)
-        } catch (t: Throwable) {
+            val proposal =
+                backend.proposeShielding(usk.account, SHIELDING_THRESHOLD, memo)
+                    ?: throw SdkException(
+                        "Insufficient balance (have 0, need $SHIELDING_THRESHOLD including fee)",
+                        null
+                    )
+            val transactionIds = backend.createProposedTransactions(proposal, usk)
+            assert(transactionIds.size == 1)
+            transactionIds[0]
+        }.onFailure {
             // TODO [#680]: if this error matches: Insufficient balance (have 0, need 1000 including fee)
             //  then consider custom error that says no UTXOs existed to shield
             // TODO [#680]: https://github.com/zcash/zcash-android-wallet-sdk/issues/680
-            Twig.debug(t) { "Shield failed" }
-            throw t
-        }.also { result ->
+            Twig.error(it) { "Shield failed" }
+        }.onSuccess { result ->
             Twig.debug { "result of shieldToAddress: $result" }
-        }
+        }.getOrThrow()
+    }
+
+    companion object {
+        private const val SHIELDING_THRESHOLD = 100000L
```

Comment:
This is 0.001 ZEC, currently worth 3.4 US cents. Because the minimum ZIP 317 fee is 10000 zats or 0.0001 ZEC, up to 10% (maybe only 5%, not sure) of the value could be lost in fees if people are sending you micropayments.

Reply:
This constant is unaltered from the prior behaviour, and the method this is in will be removed in 2.1.0.